src/test/ui/typeck/assign-non-lval-needs-deref.stderr

```diff
@@ -0,0 +1,16 @@
+error[E0067]: invalid left-hand side of assignment
+  --> $DIR/assign-non-lval-needs-deref.rs:13:16
+   |
+LL |     (&mut Foo) += ();
+   |     ---------- ^^
+   |     |
+   |     cannot assign to this expression
+   |
+help: consider dereferencing the left-hand side of this operation
+   |
+LL |     *(&mut Foo) += ();
```

Comment:
This suggestion may be a bit strange here in this minimized, but it makes sense if `&mut Foo` is coming from a function argument or something.

Reply:
FWIW, it looks like a function with a return type of `&mut Foo` can also trigger this ICE. `foo() += ();`